### ISSUE TITLE
reverse-cont fails with Assertion `t->regs().ip() - callregs.ip() == syscall_instruction_length(t->arch())' failed to hold. Should have advanced ip by one syscall_insn

### ISSUE THREAD

**Issue by mstange** [mozilla/rr#1450]
I'm using Jeff's machine in the Toronto office to debug a Firefox debug build with rr and reverse-continue is hitting an assertion. Benoit told me that you (roc) already have access to that machine.
I don't have any more time to debug this today, but I can play around with it tomorrow and see what's necessary to reproduce the bug.
I'm on rr revision a2e2c5202d3439773199424e11f886a48177ddf9 .

```
[...]
(gdb) b FrameLayerBuilder.cpp:3736
Breakpoint 1 at 0x2aaab13e655b: file /home/mstange/code/mozilla-central/layout/base/FrameLayerBuilder.cpp, line 3736.
(gdb) condition 1 item == 0x2aaad5e138f8
Breakpoint 1, mozilla::ContainerState::ProcessDisplayItems (this=0x7ffffffee650, aList=0x2aaad5e13a90) at /home/mstange/code/mozilla-central/layout/base/FrameLayerBuilder.cpp:3736
(gdb) c
Continuing.
[...]
(gdb) b mozilla::PaintedLayerDataNode::Finish
Breakpoint 2 at 0x2aaab13e29e7: file /home/mstange/code/mozilla-central/layout/base/FrameLayerBuilder.cpp, line 2432.
(gdb) condition 2 mAnimatedGeometryRoot == 0x2aaade68b548
(gdb) reverse-cont
Continuing.
[FATAL /home/mstange/code/rr/rr/src/AutoRemoteSyscalls.cc:112:syscall_helper() errno: 0 'Success'] 
 (task 23916 (rec:10965) at time 73525)
 ->  Assertion `t->regs().ip() - callregs.ip() == syscall_instruction_length(t->arch())' failed to hold. Should have advanced ip by one syscall_insn
Launch gdb with
  gdb /home/mstange/code/obj-m-debug/dist/bin/firefox
and attach to the rr debug server with:
  target remote :23916
```

**Comment by rocallahan** [comment]
I have access to the machine under UID `bgirard`, but it would be helpful to have access to the actual trace you're trying to replay as well as the exact steps to reproduce during replay of that trace. Thanks!

**Comment by mstange** [comment]
I made a mistake and asked Jeff to install some updates, which changed system libraries and made the trace invalid. I got a new trace and hit the bug a few more times, but when I tried to reproduce it following the same steps, I didn't hit it. I'll let you know when I hit it again.
Jeff gave you access to my account on the machine (mstange) by copying your public key from Benoit's .ssh/authorized_keys.
